examples/batch/fargatecomputeenvironment.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: batch.aws.crossplane.io/v1alpha1
+kind: ComputeEnvironment
+metadata:
+  name: example
+spec:
+  forProvider:
+    region: us-east-1
+    type_: MANAGED
+    computeEnvironmentName: examplecomputeEnvironmentName
+    ## Defined in examples/iam
+    #serviceRoleRef:
+    #  name: batch-role
+    ## If no service Role provided, AWS will auto-create a Service-linked-Role (or use a existing one, if there is one) for type MANAGED!
+    # ComputeResources required for type MANAGED (required/accepted fields differ from type)
+    computeResources:
+      type_: FARGATE
```

Comment:
Seems to be that `type` has a special meaning and we need to use `type_`.

Reply:
In other IAC Tools we see only type instead of type_

Reply:
@wotolom ?

Reply:
I don't know really why or what ACK is doing there.
You can find the same behavior in other Services/Resources:

- APIGateway: [zz_authorizer.go:90](https://github.com/crossplane-contrib/provider-aws/blob/f0fa1f4db6bfa1eba3edca961fd4da018aabf8d1/apis/apigateway/v1alpha1/zz_authorizer.go#L90) and [zz_integration.go:106](https://github.com/crossplane-contrib/provider-aws/blob/f0fa1f4db6bfa1eba3edca961fd4da018aabf8d1/apis/apigateway/v1alpha1/zz_integration.go#L106) (looks like for APIGatewayv2 theses fields got a new name in/through the AWS-SDK ("authorizerType", "integrationType"))
- ELBv2: [custom_types.go:107](https://github.com/crossplane-contrib/provider-aws/blob/f0fa1f4db6bfa1eba3edca961fd4da018aabf8d1/apis/elbv2/v1alpha1/custom_types.go#L107) and [:138](https://github.com/crossplane-contrib/provider-aws/blob/f0fa1f4db6bfa1eba3edca961fd4da018aabf8d1/apis/elbv2/v1alpha1/custom_types.go#L138) (here they got renamed by the crossplane-contributor)
- Batch: [zz_compute_environment.go:50](https://github.com/crossplane-contrib/provider-aws/blob/65f9d0cd440e26fa6bedee9e1e7e6a5aac058227/apis/batch/v1alpha1/zz_compute_environment.go#L50), [zz_types.go:131](https://github.com/crossplane-contrib/provider-aws/blob/65f9d0cd440e26fa6bedee9e1e7e6a5aac058227/apis/batch/v1alpha1/zz_types.go#L131)([:89](https://github.com/crossplane-contrib/provider-aws/blob/65f9d0cd440e26fa6bedee9e1e7e6a5aac058227/apis/batch/v1alpha1/zz_types.go#L89) another occurence, but which is not relevant for us)
(in the manual written Resources (Job, JobDefinition) I sometimes used the opportunity to rename the fields to something other than just "type", even that it was not really necessary: [common_types.go:41](https://github.com/crossplane-contrib/provider-aws/blob/65f9d0cd440e26fa6bedee9e1e7e6a5aac058227/apis/batch/manualv1alpha1/common_types.go#L41) (renamed), [jobdefinition.go:761](https://github.com/crossplane-contrib/provider-aws/blob/65f9d0cd440e26fa6bedee9e1e7e6a5aac058227/apis/batch/manualv1alpha1/jobdefinition.go#L761) (renamed), [job.go:149](https://github.com/crossplane-contrib/provider-aws/blob/65f9d0cd440e26fa6bedee9e1e7e6a5aac058227/apis/batch/manualv1alpha1/job.go#L149) (no "rename", oversight from me))

So a common factor seems to be that we got in one Service 2+ fields with the name "type" in 1 or 2 Resources, or in 2 or more different structs.

The reason why I did not rename the [zz_type.go:131](https://github.com/crossplane-contrib/provider-aws/blob/65f9d0cd440e26fa6bedee9e1e7e6a5aac058227/apis/batch/v1alpha1/zz_types.go#L131) one, was because I was unsure if the renaming of one field would be worth making/moving the field from the ComputeResource-struct to the CustomComputeEnvironmentParameters-struct. 
As in APIGateway (v1) the fields are implemented with json "type_", I thought it may be considered only a minor nuisance.

I do not really like the look of it neither and I may just have settled with it, because I was confronted with it from my first contribution on. ;)

PS: If you think going for the renaming is a good idea, I would probably want to use the opportunity to simple remake the entire ComputeResource-struct in custom_types.go. There are already some fields I moved out of that into CustomComputeEnvironmentParameters to be able to implement Refs/Selectors.